clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/PubSubBasedAdminClientManager.java

```diff
@@ -116,100 +103,83 @@ private Future<TopicAdminClient> createTopicAdminClient() {
         }
     }
 
-    /**
-     * Gets an existing or creates a new Topic and Subscription on Pub/Sub based on the given topic endpoint and prefix.
-     *
-     * @param endpoint The endpoint name of the topic and the subscription, e.g. command_internal.
-     * @param prefix The prefix of the topic and the subscription, e.g. the adapter instance ID.
-     * @return A succeeded Future if topic and subscription are successfully created or already exists, or a failed
-     *         Future if the topic or the subscription could not be created.
-     */
-    public Future<String> getOrCreateTopicAndSubscription(final String endpoint, final String prefix) {
-        final String topicAndSubscriptionName = PubSubMessageHelper.getTopicName(endpoint, prefix);
-        final TopicName topic = TopicName.of(projectId, topicAndSubscriptionName);
-        final SubscriptionName subscription = SubscriptionName.of(projectId, topicAndSubscriptionName);
-
-        return getOrCreateTopic(topic)
-                .onFailure(t -> LOG.warn("Error creating topic {}", topicAndSubscriptionName))
-                .recover(Future::failedFuture)
-                .compose(s -> getOrCreateSubscription(subscription, topic));
-    }
-
     /**
      * Closes the TopicAdminClient and the SubscriptionAdminClient if it exists. This method is expected to be invoked
      * as soon as the TopicAdminClient and the SubscriptionAdminClient is no longer needed.
      *
      * @return A future that is completed when the close operation completed or a succeeded future if no
      *         TopicAdminClient or SubscriptionAdminClient existed.
-     * @throws IllegalStateException if clients are not running on a Vert.x Context.
      */
     public Future<Void> closeAdminClients() {
         if (topicAdminClient == null && subscriptionAdminClient == null) {
             return Future.succeededFuture();
         }
-        final Promise<Void> adminClientClosePromise = Promise.promise();
-        if (context == null) {
-            throw new IllegalStateException("Clients are not running on a Vert.x Context");
-        } else {
-            context.executeBlocking(future -> {
-                closeSubscriptionAdminClient();
-                closeTopicAdminClient();
-                future.complete();
-            }, adminClientClosePromise);
-            return adminClientClosePromise.future();
-        }
+        closeSubscriptionAdminClient();
```

Comment:
this should definitely not be done on the event loop as both calls might take up to five seconds ...
so, using Context.executeBlocking() is the right thing to do FMPOV ...

Reply:
but I would make the constructor's vertx parameter mandatory, thus making the null check here obsolete...

Reply:
I see. In that case I would propose to let the method simply return void in order to prevent getting the impression that there is something being done asynchronously. You should then also add a note to the JavaDoc indicating that this method will block the current thread for up to 10 seconds. Any caller can then decide how to deal with this ...